source/MaterialXRuntime/Private/PvtStage.cpp

```diff
@@ -88,14 +88,63 @@ void PvtStage::removePrim(const PvtPath& path)
         throw ExceptionRuntimeError("Given path '" + path.asString() + " does not point to a prim in this stage");
     }
 
+    // Destroy this prim and all its children.
+    prim->destroy();
+
+    // Remove from its parent.
+    PvtPrim* parent = prim->getParent();
+    parent->removeChildPrim(prim);
+}
+
+void PvtStage::disposePrim(const PvtPath& path)
+{
+    PvtPrim* prim = getPrimAtPathLocal(path);
+    if (!(prim && prim->getParent()))
+    {
+        throw ExceptionRuntimeError("Given path '" + path.asString() + " does not point to a prim in this stage");
+    }
+
+    // Make sure the prim has no connections.
+    for (RtAttribute attr : prim->getAttributes())
```

Comment:
Make it an error to dispose a prim with existing connections. Connections should be broken first. To make undo/redo handling easier.

Reply:
This is good to have. Should catch any code which incrementally tries and delete w/o disconnect which don't use commands.